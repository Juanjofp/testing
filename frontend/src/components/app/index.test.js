import React from 'react';
import ReactDOM from 'react-dom';
import {render} from 'enzyme';
import App from './index';

window.__MUI_USE_NEXT_TYPOGRAPHY_VARIANTS__ = true;

describe(
    'App component',
    () => {
        test(
            'renders without crashing',
            () => {
                const div = document.createElement('div');
                ReactDOM.render(<App isLogin={true} isAdmin={true}/>, div);
            }
        );

        test(
            'create an snapshot',
            () => {
                const tree = render(<App />);
                expect(tree).toMatchSnapshot();
            }
        );
    }
);
