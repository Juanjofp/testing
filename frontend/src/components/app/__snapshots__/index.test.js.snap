// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component create an snapshot 1`] = `
<div>
  <header
    class="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-1 MuiAppBar-positionSticky-4 MuiAppBar-colorPrimary-8"
  >
    <div
      class="MuiToolbar-root-37 MuiToolbar-regular-39 MuiToolbar-gutters-38"
    >
      <button
        aria-label="Open drawer"
        class="MuiButtonBase-root-47 MuiIconButton-root-41 MuiIconButton-colorInherit-42"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label-46"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-50"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
      </button>
      <h6
        class="MuiTypography-root-59 MuiTypography-h6-76 MuiTypography-colorInherit-88"
      >
        Centic Testing
      </h6>
    </div>
  </header>
  <div>
    <div
      style="padding-top:64px;height:90vh"
    >
      <div
        data-testid="public-main"
      >
        <div>
          <button
            data-testid="home-button"
          >
            0
          </button>
          <div>
            <p>
              Componente hola
            </p>
          </div>
          <div>
            <p>
              Componente hola
            </p>
          </div>
          <div>
            <p>
              Componente hola
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
